_posts/2017-07-03-Devoxx-pl.md

```diff
@@ -165,4 +172,203 @@ With tools like [Split](https://www.split.io) and [LaunchDarkly](https://launchd
+Another improvement is using immutable objects wherever possible.
+Using a `StringBuilder` instead of concatenating Strings will reduce heap usage. 
+For more info, Oleg referred to a talk on [Moving to G1GC by Kirk Pepperdine](https://vimeo.com/221269942)
+
+### Http2 Client
+The JDK 9 will contain an incubator package with a Http2 client with a fluent API:
+
+{% highlight Java %}
+HttpClient.newHttpClient().send(
+        newBuilder(URI.create("https://google.com"))
+            .GET()
+            .build(),
+        HttpResponse.BodyHandler.asString())
+    .body();
+)
+{% endhighlight %}
+
+Oleg concluded by recommending the audience not to touch multi-release jars, jlink and Unsafe, unless you are 100% sure what you are doing.
+For now he recommends to upgrade your IDE and tools and upgrade Spring to version 5.0. 
+Then add the `--illegal-access=warn` startup option and fix the easy fixable warnings and then wait a year or more until the classpath and the libraries you depend on are upgraded.
+
+Yannick, a colleague at Ordina did a nice presentation on Java 9: [A first look at Java 9](https://www.youtube.com/watch?v=StJPI2rUPfs) by [Yannick De Turck](https://twitter.com/@YannickDeTurck).
+Presentation and sources are available at [GitHub](https://github.com/yannickdeturck/java9workshop).
+
+## The Language of Actors. ([Vaughn Vernon](https://twitter.com/@VaughnVernon) - [Slides](https://github.com/VaughnVernon/DDDwithActors))
+
+Vaughn started his talk by introducing Rear Admiral Grace Hopper to the audience.
+In the American Navy, she was a Computer Scientist and wrote software for a long time.
+She was much into not wasting cycles and emphasised on not wasting nanoseconds.
+
+Then Vaughn introduced Donald Knuth, another legend in Computer Science.
+Knuth is known from the quote _Premature optimization is the root of all evil_.
+But that is not exactly what he said, the full quote says:
+
+> We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. 
+Yet we should not pass up our opportunities in that critical 3%.
+
+Another quote was shown us from Donald Knuth:
+
+>People who are more than casually interested in computers should have at least some idea of what the underlying hardware is like. 
+Otherwise the programs they write will be pretty weird.
+
+To further build his point Vaughn told about a project that was written in Cobol.
+The code was across 5 diskettes thus user interaction was needed to run the application.
+To improve the usability of the application, it was rewritten in C, allowing the software to fit on just one diskette.
+With this introduction Vaughn wanted to emphasize how hard it is to optimize software for resource usage.
+
+### Threading is hard
+
+In 1973 academics discovered the **Actor Model**, 13 year later in 1986 Joe Armstrong rediscovered the approach.
```

Comment:
In 1973, (comma) academics discovered the **Actor Model**. (period)
13 year later in 1986 Joe Armstrong rediscovered the approach.